nomenclature/testing.py

```diff
@@ -55,6 +56,8 @@ def assert_valid_structure(
         - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
 
     """
+    if dimensions == []:
+        dimensions = os.listdir(f"{str(path)}/{definitions}")
```

Comment:
This one's a bit dangerous as it lists _all_ folders and _files_ in the definitions directory. Also you are mixing functions from the `os` module and from `pathlib`. The latter is the newer way to do things and almost everything that the `os` module implements in terms of file operations can be achieved with `pathlib`. Additionally, you don't have to convert to string anymore. Here's a list of how the `pathlib` module implements the `os` module's functionality, https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module.
```suggestion
        dimensions = [x for x in (path / definitions).iterdir() if x.is_dir()]
```

Reply:
@phackstock The os version allowed me to directly get the name of the directories. With the option you suggest, we get wholes paths and I have to change what is then sent to the next variable. But OK, I can try to work with that.

Reply:
Ah yes sorry that was a little imprecise on my side. This should do the job: 
```suggestion
       dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
```
Reference: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem